### ISSUE TITLE
Bravo 1.2 won't start on Dreamhost Linux VPS (Debian Linux 5.0.8)

### ISSUE THREAD

**Issue by Frankinator** [bravoserver/bravo#149]
I'm trying to start Bravo on a Dreamhost VPS running Debian Linux 5.0.8. I followed the instructions on the GitHub page (install twisted and numpy, rename the ini, etc.) but when I run

```
twistd -ny bravo.tac
```

It shoots out this:

[ps46000]$ twistd -ny bravo.tac
Traceback (most recent call last):
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/application/app.py", line 631, in run
    runApp(config)
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/scripts/twistd.py", line 23, in runApp
    _SomeApplicationRunner(config).run()
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/application/app.py", line 374, in run
    self.application = self.createOrGetApplication()
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/application/app.py", line 439, in createOrGetApplication
    application = getApplication(self.config, passphrase)
--- <exception caught here> ---
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/application/app.py", line 450, in getApplication
    application = service.loadApplication(filename, style, passphrase)
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/application/service.py", line 400, in loadApplication
    application = sob.loadValueFromFile(filename, 'application', passphrase)
  File "/usr/lib/python2.5/site-packages/Twisted-10.2.0-py2.5-linux-x86_64.egg/twisted/persisted/sob.py", line 210, in loadValueFromFile
    exec fileObj in d, d
  File "bravo.tac", line 1, in <module>
    from bravo.service import application
  File "/home/solidinc_minec/mymcdirectory/bravo/service.py", line 7, in <module>
    from bravo.factories.beta import BravoFactory
  File "/home/solidinc_minec/mymcdirectory/bravo/factories/beta.py", line 15, in <module>
    from bravo.protocols.beta import BravoProtocol
  File "/home/solidinc_minec/mymcdirectory/bravo/protocols/beta.py", line 37, in <module>
    BuildData = namedtuple("BuildData", "block, metadata, x, y, z, face")
  File "/home/solidinc_minec/mymcdirectory/bravo/compat.py", line 166, in namedtuple
    raise SyntaxError(e.message + ':\n' + template)
exceptions.SyntaxError: :
class BuildData(tuple):
            'BuildData(block, metadata, x, y, z, face)'

```
        __slots__ = ()

        _fields = ('block', 'metadata', 'x', 'y', 'z', 'face')

        def __new__(_cls, block, metadata, x, y, z, face):
            return _tuple.__new__(_cls, (block, metadata, x, y, z, face))

        @classmethod
        def _make(cls, iterable, new=tuple.__new__, len=len):
            'Make a new BuildData object from a sequence or iterable'
            result = new(cls, iterable)
            if len(result) != 6:
                raise TypeError('Expected 6 arguments, got %d' % len(result))
            return result

        def __repr__(self):
            return 'BuildData(block=%r, metadata=%r, x=%r, y=%r, z=%r, face=%r)' % self

        def _asdict(self):
            'Return a new dict which maps field names to their values'
            return dict(zip(self._fields, self))

        def _replace(_self, **kwds):
            'Return a new BuildData object replacing specified fields with new values'
            result = _self._make(map(kwds.pop, ('block', 'metadata', 'x', 'y', 'z', 'face'), _self))
            if kwds:
                raise ValueError('Got unexpected field names: %r' % kwds.keys())
            return result

        def __getnewargs__(self):
            return tuple(self)

    block = _property(_itemgetter(0))
    metadata = _property(_itemgetter(1))
    x = _property(_itemgetter(2))
    y = _property(_itemgetter(3))
    z = _property(_itemgetter(4))
    face = _property(_itemgetter(5))
```

Failed to load application: :
class BuildData(tuple):
            'BuildData(block, metadata, x, y, z, face)'

```
        __slots__ = ()

        _fields = ('block', 'metadata', 'x', 'y', 'z', 'face')

        def __new__(_cls, block, metadata, x, y, z, face):
            return _tuple.__new__(_cls, (block, metadata, x, y, z, face))

        @classmethod
        def _make(cls, iterable, new=tuple.__new__, len=len):
            'Make a new BuildData object from a sequence or iterable'
            result = new(cls, iterable)
            if len(result) != 6:
                raise TypeError('Expected 6 arguments, got %d' % len(result))
            return result

        def __repr__(self):
            return 'BuildData(block=%r, metadata=%r, x=%r, y=%r, z=%r, face=%r)' % self

        def _asdict(self):
            'Return a new dict which maps field names to their values'
            return dict(zip(self._fields, self))

        def _replace(_self, **kwds):
            'Return a new BuildData object replacing specified fields with new values'
            result = _self._make(map(kwds.pop, ('block', 'metadata', 'x', 'y', 'z', 'face'), _self))
            if kwds:
                raise ValueError('Got unexpected field names: %r' % kwds.keys())
            return result

        def __getnewargs__(self):
            return tuple(self)

    block = _property(_itemgetter(0))
    metadata = _property(_itemgetter(1))
    x = _property(_itemgetter(2))
    y = _property(_itemgetter(3))
    z = _property(_itemgetter(4))
    face = _property(_itemgetter(5))
```

Which returns nothing of use to me. The server hasn't started, as trying to connect gets me a connection refused error. I've tried running as root (no luck) and also ran the setup.py (which again did nothing).

Is there something else I need to run? Something I've missed?

**Comment by MostAwesomeDude** [comment]
898736fd4e56677e3830be5a5b23c4d5a0e47f83 fixed this, however, it's not in 1.2. Yet more incentive for me to release 1.3, especially since I just unbroke a wall of tests.

Also, DH charges by RAM usage for their VPS, and you might find it a bit spendy for your tastes. OTOH, it's hard to find a good VPS for this kind of thing.

**Comment by Frankinator** [comment]
That fixed it up nicely. For anyone else reading, I just replaced the contents of compat.py with 898736f. Then I ran  

```
twistd -ny bravo.tac
```

again and now I'm happily minecrafting.

As for the DH thing, Bravo is only using 100mb of memory. Compare that to the 1gb that the official MC server is requesting (and the 300mb limit that I gave it to pinch a few pennies!), plus I'm also planning to host several client sites on the VPS, so Bravo might be the winner in this.

Thanks for the prompt support!
